build/CompilationFlags.cmake

```diff
@@ -20,7 +20,9 @@ if("${CMAKE_CXX_COMPILER_ID}" MATCHES "^(AppleClang|Clang|GNU)$")
     add_flag(-Wno-unused-function) # prints too many useless warnings
     add_flag(-Wno-format-nonliteral) # prints way too many warnings from spdlog
     add_flag(-Wno-gnu-zero-variadic-macro-arguments) # https://stackoverflow.com/questions/21266380/is-the-gnu-zero-variadic-macro-arguments-safe-to-ignore
-
+	add_flag(-Wno-unused-result) #Every logger call generates this
```

Comment:
Disabling unused result might hide some bad practices. Do we have a way to change the logger to avoid this? If not, it's fine then.